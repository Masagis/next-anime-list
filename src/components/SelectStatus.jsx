import React from 'react'
import Select from 'react-select'

function SelectStatus(props) {
  const { handleFilter, statusType } = props

  return (
    <div className="col-md-4 mb-2">
      <Select
        data-testid="type-status-anime"
        name="status"
        placeholder="- Filter Status -"
        options={statusType}
        onChange={(e) => handleFilter(e?.value, 'status')}
      />
    </div>
  )
}

export default SelectStatus
