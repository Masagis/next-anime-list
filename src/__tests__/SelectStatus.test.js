/* eslint-disable no-undef */
import { fireEvent, render } from '@testing-library/react'
import SelectStatus from '../components/SelectStatus'

describe('Component SelectStatus', () => {
  test('should be render the component', () => {
    const statusType = [
      {
        value: '',
        label: '- All Status-',
      },
      {
        value: 'airing',
        label: 'Airing',
      },
      {
        value: 'complete',
        label: 'Complete',
      },
      {
        value: 'upcoming',
        label: 'Upcoming',
      },
    ]

    const handleFilter = jest.fn()
    const { container, getByText } = render(
      <SelectStatus statusType={statusType} handleFilter={handleFilter} />,
    )

    fireEvent.focus(container.querySelector('input'))
    fireEvent.keyDown(container.querySelector('input'), {
      key: 'ArrowDown',
      code: 40,
    })
    fireEvent.click(getByText('Complete'))

    expect(handleFilter).toHaveBeenCalledWith('complete', 'status')
  })

  test('snapshoot', () => {
    const statusType = [
      {
        value: '',
        label: '- All Status-',
      },
      {
        value: 'airing',
        label: 'Airing',
      },
      {
        value: 'complete',
        label: 'Complete',
      },
      {
        value: 'upcoming',
        label: 'Upcoming',
      },
    ]
    const handleFilter = jest.fn()
    const renderComponent = render(
      <SelectStatus statusType={statusType} handleFilter={handleFilter} />,
    )
    expect(renderComponent).toMatchSnapshot()
  })
})
