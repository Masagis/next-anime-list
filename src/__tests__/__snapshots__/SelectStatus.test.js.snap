// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SelectStatus snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md-4 mb-2"
      >
        <div
          class=" css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-14el2xx-placeholder"
                id="react-select-3-placeholder"
              >
                - Filter Status -
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="status"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md-4 mb-2"
    >
      <div
        class=" css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-14el2xx-placeholder"
              id="react-select-3-placeholder"
            >
              - Filter Status -
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          name="status"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
